src/wallet/rpcwallet.cpp

```diff
@@ -832,48 +832,74 @@ static UniValue getbalance(const JSONRPCRequest& request)
+                "3. include_watchonly (bool, optional, default=false) DEPRECATED. Only valid when an account is specified. This argument will be removed in V0.18. To use this deprecated argument, start bitcoind with -deprecatedrpc=accounts. Also include balance in watch-only addresses (see 'importaddress')\n"
+                "\nResult:\n"
+                "amount              (numeric) The total amount in " + CURRENCY_UNIT + " received for this account.\n"
+                "\nExamples:\n"
+                "\nThe total amount in the wallet with 1 or more confirmations\n"
+                + HelpExampleCli("getbalance", "") +
+                "\nThe total amount in the wallet at least 6 blocks confirmed\n"
+                + HelpExampleCli("getbalance", "\"*\" 6") +
+                "\nAs a json rpc call\n"
+                + HelpExampleRpc("getbalance", "\"*\", 6")
+            );
+        } else {
+            throw std::runtime_error(
+                "getbalance ( include_unavailable )\n"
+                "\nReturns the server's total available balance.\n"
+                "The available balance is what the wallet considers currently spendable, and is\n"
+                "thus affected by options which limit spendability such as -spendzeroconfchange.\n"
+                "\nArguments:\n"
+                "1. include_unavailable (bool, optional, default=false) Also include balance for UTXOs which are currently not (but possibly in the future)\n"
+                "                                                       spendable.\n"
+                "\nResult:\n"
+                "amount              (numeric) The total amount in " + CURRENCY_UNIT + " received.\n"
+                "\nExamples:\n"
+                "\nThe total amount in the wallet\n"
+                + HelpExampleCli("getbalance", "") +
+                "\nThe total amount in the wallet, including currently unspendable UTXO:s\n"
+                + HelpExampleCli("getbalance", "true") +
+                "\nAs a json rpc call\n"
+                + HelpExampleRpc("getbalance", "true")
+            );
+        }
+    }
 
     // Make sure the results are valid at least up to the most recent block
     // the user could have gotten from another RPC command prior to now
     pwallet->BlockUntilSyncedToCurrentChain();
 
     LOCK2(cs_main, pwallet->cs_wallet);
 
+    // params[0] is not converted from JSON until account stuff is removed, so we need to manually convert it here
+    // TODO: add param[0] to clients.cpp for getbalance and switch to get_bool() here, once account stuff is removed
+    const bool include_unavailable = !IsDeprecatedRPCEnabled("accounts") && !request.params[0].isNull() && (request.params[0].get_str() == "1" || request.params[0].get_str() == "true");
```

Comment:
It's a bit ugly to replicate the univalue parsing here, but I think it's ok since it's only going to be in here for one release.

I think you should remove the `request.params[0].get_str() == "1"` (1 isn't interpreted as true in univalue). You may also throw an error if `request.params[0].get_str()` is not `false` or `true` (so passing `True` for example doesn't silently fail and leave `include_unavailable` as false.

Reply:
I thought `1` was true. E.g. `getrawtransaction <id> 1`. But that seems to be a special case for `getrawtransaction`, actually. Changing as you suggest.